util/file_reader_writer.h

```diff
@@ -138,7 +138,9 @@ class WritableFileWriter {
         rate_limiter_(options.rate_limiter),
         stats_(stats) {
     buf_.Alignment(writable_file_->GetRequiredBufferAlignment());
-    buf_.AllocateNewBuffer(std::min((size_t)65536, max_buffer_size_));
+    buf_.AllocateNewBuffer(use_direct_io()
+                               ? max_buffer_size_
+                               : std::min((size_t)65536, max_buffer_size_));
```

Comment:
@siying This incremental increase approach was there before I came in. The purpose was to save on a resident memory, I guess. It is fine to make it fixed in general.